ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/ExplicitDisclosureIT.scala

```diff
@@ -8,98 +8,218 @@ import com.daml.ledger.api.testtool.infrastructure.Allocation._
 import com.daml.ledger.api.testtool.infrastructure.Assertions._
 import com.daml.ledger.api.testtool.infrastructure.LedgerTestSuite
 import com.daml.ledger.api.testtool.infrastructure.TransactionHelpers.createdEvents
+import com.daml.ledger.api.testtool.infrastructure.participant.ParticipantTestContext
 import com.daml.ledger.api.v1.commands.DisclosedContract
 import com.daml.ledger.api.v1.event.CreatedEvent
 import com.daml.ledger.client.binding
 import com.daml.ledger.test.model.Test._
 
+import scala.concurrent.{ExecutionContext, Future}
+
 final class ExplicitDisclosureIT extends LedgerTestSuite {
+  import ExplicitDisclosureIT._
 
   test(
-    "EDPlaceholder",
-    "Placeholder test (only to check whether it compiles)",
-    allocate(Parties(2)),
-  )(implicit ec => { case Participants(Participant(ledger, owner, delegate)) =>
-    val contractKey = "SomeKeyValue"
-
-    def fetchDelegatedRequest(
-        delegationCid: binding.Primitive.ContractId[Delegation],
-        delegatedCid: binding.Primitive.ContractId[Delegated],
-        disclosedContract: Option[DisclosedContract],
-    ) = ledger
-      .submitAndWaitRequest(
-        delegate,
-        delegationCid.exerciseFetchDelegated(delegate, delegatedCid).command,
+    "EDMetadata",
+    "All create events have metadata defined",
+    allocate(Parties(1)),
+  )(implicit ec => { case Participants(Participant(ledger, owner)) =>
+    val contractKey = ledger.nextKeyId()
+    for {
+      _ <- ledger.create(owner, Delegated(owner, contractKey))
+      flatTx <- ledger.flatTransactions(owner)
+      tree <- ledger.transactionTrees(owner)
+      treeById <- ledger.transactionTreeById(flatTx.head.transactionId, owner)
+    } yield {
+      assert(
+        createdEvents(flatTx.head).head.metadata.isDefined,
```

Comment:
Why is the assertion checking only the head rather than all events? If you want to check that there is one or at least one event, maybe first add an assertion for that and then check all events?

Reply:
Good point, there's no need to restrict the check to the first event.